source/common/quic/quic_stat_names.cc

```diff
@@ -38,6 +42,20 @@ void QuicStatNames::chargeQuicConnectionCloseStats(Stats::Scope& scope,
                      connection_close});
 }
 
+void QuicStatNames::chargeQuicResetStreamErrorStats(Stats::Scope& scope,
+                                                    quic::QuicRstStreamErrorCode error_code,
+                                                    bool from_self, bool is_upstream) {
+  ASSERT(&symbol_table_ == &scope.symbolTable());
+
+  if (error_code > quic::QUIC_STREAM_LAST_ERROR) {
+    error_code = quic::QUIC_STREAM_LAST_ERROR;
+  }
+
+  const Stats::StatName stream_error = resetStreamErrorStatName(error_code);
+  incCounter(scope, {http3_prefix_, is_upstream ? upstream_ : downstream_,
+                     from_self ? from_self_ : from_peer_, stream_error});
```

Comment:
nit: Up to you, but I think I would put ()s around the two "?:" expressions:

  incCounter(scope, {http3_prefix_, (is_upstream ? upstream_ : downstream_),
                     (from_self ? from_self_ : from_peer_), stream_error});